Library/Homebrew/dependency.rb

```diff
@@ -71,25 +71,30 @@ class << self
     # The default filter, which is applied when a block is not given, omits
     # optionals and recommendeds based on what the dependent has asked for.
     def expand(dependent, deps = dependent.deps, &block)
+      @expand_stack ||= []
+      @expand_stack.push dependent
+
       expanded_deps = []
 
       deps.each do |dep|
-        # FIXME: don't hide cyclic dependencies
         next if dependent.name == dep.name
 
         case action(dependent, dep, &block)
         when :prune
           next
         when :skip
+          next if @expand_stack.include? dependent
```

Comment:
> As a quick sanity check, brew uses --recursive <formula> should always return a longer or equally long list of formulae when compared to brew uses <formula>. With the current code, it doesn't.

@UniqMartin Sorry, I'm not totally sure I understand how these two issues are related?